target/linux/ramips/dts/mt7621_tplink_er605-v2.dts

```diff
@@ -0,0 +1,208 @@
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_STATUS;
+			gpios = <&gpio 7 GPIO_ACTIVE_HIGH>;
+			default-state = "keep";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset button";
+			gpios = <&gpio 8 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	reg_usb_vbus: regulator-usb {
+		compatible = "regulator-fixed";
+		regulator-name = "usb_vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		gpio = <&gpio 10 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+
+	reg_3p3v: regulator-3p3v {
+		compatible = "regulator-fixed";
+		regulator-name = "fixed-3.3V";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+};
+
+&gmac0 {
+	nvmem-cells = <&macaddr_factory_22>;
+	nvmem-cell-names = "mac-address";
+	label = "dsa";
+};
+
+&gmac1 {
+	status = "okay";
+	label = "eth0";
+	phy-handle = <&ethphy0>;
+
+	nvmem-cells = <&macaddr_factory_22>;
+	nvmem-cell-names = "mac-address";
+};
+
+
+&switch0 {
+	ports {
+		port@0 {
+			status = "disabled";
+		};
+
+		port@1 {
+			status = "okay";
+			label = "eth1";
+			nvmem-cells = <&macaddr_factory_22>;
+			nvmem-cell-names = "mac-address";
+			mac-address-increment = <1>;
+		};
+
+		port@2 {
+			status = "okay";
+			label = "eth2";
+			nvmem-cells = <&macaddr_factory_22>;
+			nvmem-cell-names = "mac-address";
+			mac-address-increment = <2>;
+		};
+
+		port@3 {
+			status = "okay";
+			label = "eth3";
+			nvmem-cells = <&macaddr_factory_22>;
+			nvmem-cell-names = "mac-address";
+			mac-address-increment = <3>;
+		};
+
+		port@4 {
+			status = "okay";
+			label = "eth4";
+			nvmem-cells = <&macaddr_factory_22>;
+			nvmem-cell-names = "mac-address";
+			mac-address-increment = <4>;
+		};
+	};
+};
+
+&nand {
+	status = "okay";
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "u-boot";
+			reg = <0x0 0x80000>;
+			read-only;
+		};
+
+		partition@80000 {
+			label = "u-boot-env";
+			reg = <0x80000 0x80000>;
+			read-only;
+		};
+
+		factory: partition@100000 {
+			label = "factory";
+			reg = <0x100000 0x40000>;
+			read-only;
+		};
+
+		partition@140000 {
+			label = "firmware";
+			reg = <0x140000 0x7cc0000>;
+		};
+
+		partition@7e00000 {
+			label = "panic-ops";
+			reg = <0x7e00000 0x200000>;
+		};
+	};
+};
+
+
+&mdio {
+	ethphy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+};
+
+&state_default {
+	gpio {
+		groups = "uart2", "uart3", "pcie", "jtag";
+		function = "gpio";
+	};
+};
+
+&spi0 {
+	status = "disabled";
+};
+
+&xhci {
+	vusb33-supply = <&reg_3p3v>;
+	vbus-supply = <&reg_usb_vbus>;
+};
+
+// There was a partition on a flash containing JSON with the MAC, but I no longer have access to it
+&factory {
+	compatible = "nvmem-cells";
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	macaddr_factory_22: macaddr@22 {
+		reg = <0x22 0x6>;
+	};
```

Comment:
Does this partition contain the mac address which is on the label on the box?

Reply:
We would like to use the same mac addresses like the vendor firmware to prevent mac address conflicts. The vendors should have reserved some mac addresses for each device.

Reply:
Yes, I do know that mac addr was inside one of ubi volumes. However as I have said in the comments I no longer have that - I accidentally wiped that during my experiments. Would it be ok to add support for maintaining vendor's mac in follow up PRs?

Reply:
I would skim through a dump of the factory partition just in case. The label address should be fairly early on or at an aligned position like 0xe000. 

Additionally you may have to check whether the tp-link u-boot passes through the ethernet mac address itself. You may have to add an alias in that case.

Reply:
As far as I remember (before I wiped out UBI volume) the mac address is stored in a UBI volume in json format.
Factory partition is empty on this device, `u-boot-env` is empty as well.

Reply:
Sorry, correction. I just got access to the 'stock' FW and it looks like mac address is stored in a separate UBI volume, but not as json. Instead that looks like text data containing part of openwrt config. Stock FW uses openwrt.

Reply:
If you define here the `macaddr_factory_22` node and make use of it it should also contain the mac address in normal devices. If the mac address is normally stored somewhere else please do not add these.